indexers/dbms/PostgreSQLIndexer/__init__.py

```diff
@@ -0,0 +1,158 @@
+        return self
+
+    def create_table(self):
+        """
+        Create Table with id, vecs and metas.
+        """
+
+        self.cursor.execute("select exists(select * from information_schema.tables where table_name=%s)", (self.table,))
+        if self.cursor.fetchone()[0]:
+            self.logger.info("Using existing table")
+        else:
+            try:
+                self.cursor.execute(
+                    f"""DROP TABLE IF EXISTS {self.table};
+                                    CREATE TABLE {self.table} (
+                                    ID INT PRIMARY KEY, 
+                                    VECS BYTEA, 
+                                    METAS BYTEA);"""
+                )
+                self.logger.info("Successfully table created")
+            except:
+                self.logger.error("Error while creating table!")
+
+    def add(self, ids, vecs, metas, *args, **kwargs):
+        """Insert the documents into the database.
+
+        :param ids: List of doc ids to be added
+        :param vecs: List of vecs to be added
+        :param metas: List of metas of docs to be added
+        :return record: List of Document's id added
+        """
+
+        self.cursor.execute(f"DELETE FROM {self.table}")
+        for i in range(len(ids)):
+            self.cursor.execute(
+                f"INSERT INTO {self.table} (ID, VECS, METAS) VALUES (%s, %s, %s)",
+                (ids[i], pickle.dumps(vecs[i]), pickle.dumps(metas[i])),
+            )
+        self.connection.commit()
+        self.cursor.execute(f"SELECT ID from {self.table}")
+        record = self.cursor.fetchall()
+        return record
+
+    def update(self, id, vecs, metas, *args, **kwargs):
+        """Updated document from the database.
+
+        :param ids: Id of Doc to be updated
+        :param vecs: List of vecs to be updated
+        :param metas: List of metas of docs to be updated
+        :return record: List of Document's id after update
+        """
+
+        self.cursor.execute(
+            f"UPDATE {self.table} SET VECS = %s, METAS = %s WHERE ID = %s",
+            (pickle.dumps(vecs), pickle.dumps(metas), id),
+        )
+        self.connection.commit()
+        self.cursor.execute(f"SELECT ID from {self.table}")
+        record = self.cursor.fetchall()
+        return record
+
+    def delete(self, id, *args, **kwargs):
+        """Delete document from the database.
+
+        :param id: Id of Document to be removed
+        :return record: List of Document's id after deletion
+        """
+
+        self.cursor.execute(f"DELETE FROM {self.table} where (ID) = (%s) ", id)
+        self.connection.commit()
+        count = self.cursor.rowcount
+        self.cursor.execute(f"SELECT ID from {self.table}")
+        record = self.cursor.fetchall()
+        return record
+
+    def dump(self, uri, shards, formats):
+        raise NotImplementedError
+
+    def __exit__(self, *args):
```

Comment:
the same way there is no `__enter__`. There should be no `__exit__` as this is not used in the `Runtime` as a context manager.

See how `MongoDBIndexer` uses an `inner` class as a context manager to handle the `connections` lifetime

Reply:
But Mongo does have an `__exit__` method, and this one also has an `__enter__` one, there should be none of those?

Reply:
in Mongo, is MongoDBHandler which is a context manager, not the Indexer itself. I suggest to use the same pattern to handle the connection

Reply:
Emm, yes we can do this. I removed the `__enter__` but didn't find the `__exit__` before. So `__exit__` is also not used right now. 
But it's good to use another handler. We will try this tomorrow maybe?